spring-cloud-zookeeper-config/src/main/java/org/springframework/cloud/zookeeper/config/ZookeeperConfigBootstrapConfiguration.java

```diff
@@ -42,4 +48,35 @@ public ZookeeperPropertySourceLocator zookeeperPropertySourceLocator(
 	public ZookeeperConfigProperties zookeeperConfigProperties() {
 		return new ZookeeperConfigProperties();
 	}
+
+	@Bean
+	public ApplicationListener<ContextRefreshedEvent> zookeeperTreeCachePropertySourceContextRefreshedApplicationListener() {
```

Comment:
When the property source is created, we only have access to the parent bootstrap context, not the child application context where all the real configuration lives. If there was a way to navigate down intto children contexts, then that would be sufficient.